torch/csrc/jit/script/parser.h

```diff
@@ -182,11 +185,58 @@ struct Parser {
       L.expect(end);
     return List<T>::create(r, elements);
   }
+
   Const parseConst() {
     auto range = L.cur().range;
     auto t = L.expect(TK_NUMBER);
     return Const::create(t.range, t.text());
   }
+
+  bool isCharCount(char c, const std::string& str, size_t start, int len) {
+    //count checks from [start, start + len) 
+    return start + len <= str.size() && std::count(str.begin() + start, str.begin() + start + len, c) == len; 
+  }
+
+  std::string parseLexedString(const std::string &str, size_t start) {
```

Comment:
This is a bit confusing. Why do we concatenate adjacent string literals in lexer, but we do it in a way that requires cleanup later? Either leave them separate in the lexer and do all the work in the parser, or emit nice and clean STRINGLITERAL tokens.

Reply:
There are two separate things here:

1. We need to parse the literal quotation into the string it represents. The lexer only returns ranges into the original token stream, so it is impossible for the lexer to handle this. So it has to be done here.
2. We need to handle multiple adjacent strings. I think this is best handled by representing each string as a separate token:

```
std::string parseStringLiteral() {
  std::stringstream ss;
  while(L.cur().kind == TK_STRING_LITERAL)
     ss << parseString(L.next().text());
  return ss.str();
} 
```